pinot-query-planner/src/main/java/org/apache/pinot/calcite/sql/fun/PinotOperatorTable.java

```diff
@@ -467,4 +481,83 @@ public boolean allowsFraming() {
       return false;
     }
   }
+
+  /// Pinot's custom SUM aggregation function that can aggregate on SV or MV numeric inputs. We can't simply override
+  /// the return type inference and operand type checker in {@link AggregationFunctionType} like we do for other
+  /// functions because {@link org.apache.calcite.sql.fun.SqlSumAggFunction} has some customizations that we need to
+  /// retain here to ensure that rules like {@link org.apache.calcite.rel.rules.AggregateRemoveRule} work as expected.
+  /// TODO: Replace {@link SqlStdOperatorTable#SUM} with this instance after the next release (there's a dependency
+  /// on AVG window function, see https://github.com/apache/pinot/pull/17109).
+  private static final class PinotSumFunction extends PinotSqlAggFunction {
```

Comment:
Is this used?
Should we introduce this change (also MIN and MAX) along with other broker side changes?